### ISSUE TITLE
Probrem at autoload when NeoBundleLazy is used

### ISSUE THREAD

**Issue by rcmdnk** [Shougo/neobundle.vim#409]
I'm using neobundle and [kana/vim-submode](https://github.com/kana/vim-submode),
and vimrc has codes like following lines:

```vim
  NeoBundleLazy "kana/vim-submode", {
        \ "autoload": { "commands": ["submode"]}}
...
  call submode#enter_with("winsize", "n", "", "<C-w>>", "<C-w>>")
...
```

It worked well, but with the latest neobundle, it shows

    Error detected while processing /home/user/.vimrc:
    line 1561:
    E117: Unknown function: submode#enter_with
    line 2012:
    E117: Unknown function: submode#enter_with
    Press ENTER or type command to continue

It is due to this update: [Disable autoload function feature in vim_starting · Shougo/neobundle.vim@b035e8b](https://github.com/Shougo/neobundle.vim/commit/b035e8b76b58e0dce03e6d8330b03dd011346a0b#diff-51aaef0f307e32d5911eb6c226a60ca5)

If I remove 

          \ || has('vim_starting')

then it works.

Could you please confirm if such a behaviour is intended or not?
If it is intended, I can change the configuration to such simply:

      NeoBundle "kana/vim-submode"

**Comment by Shougo** [comment]
Same issue.  https://github.com/Shougo/neobundle.vim/issues/408

You should not use vim-submode as lazy.

**Comment by arakashic** [comment]
I spotted the same problem as I commented on the commit. My case is I am using neobundle with [vim-pencil](https://github.com/reedes/vim-pencil). I only use it with certain filetype. My config is like this:
```
NeoBundleLazy "reedes/vim-pencil"
...
autocmd FileType markdown,mkd call pencil#init()
```
Although it is possbile to also add autoload option to NeoBundleLazy, it is a bad practice to manage autoload configuration for one plugin in two places.

Plus, is there anything that breaks when autoload is enabled in vim_starting?
